docs/concepts.md

```diff
@@ -0,0 +1,123 @@
+
+=== "Concept Map"
+
+    ```d2
+    --8<-- "docs/diagrams/concept-map.d2"
+    ```
+
+=== "Example for the `PrairieLearn 101` course"
+
+    ```d2
+    --8<-- "docs/diagrams/concept-map-example.d2"
+    ```
+
+### Questions
+
+[**Questions**](./question.md) are all independent, and can be graded [using Python](./question.md), or using an [external code autograder](./externalGrading.md). They are written in HTML (as a [Mustache template](https://mustache.github.io/mustache.5.html)) and generated / graded using Python.
+
+=== "Example `question.html`"
+
+    ```html title="question.html"
+    <pl-question-panel>
+        If $x = {{params.x}}$ and $y$ is {{params.operation}} $x$, what is $y$?
+    </pl-question-panel>
+
+    <pl-number-input answers-name="y" label="$y =$"></pl-number-input>
+    <pl-submission-panel> {{feedback.y}} </pl-submission-panel>
+    ```
+
+=== "Example `server.py`"
+
+    ```python title="server.py"
+    def generate(data):
+        data["params"]["x"] = random.randint(5, 10)
+        data["params"]["operation"] = random.choice(["double", "triple"])
+
+        if data["params"]["operation"] == "double":
+            data["correct_answers"]["y"] = 2 * data["params"]["x"]
+        else:
+            data["correct_answers"]["y"] = 3 * data["params"]["x"]
+
+    def parse(data):
+        if "y" not in data["format_errors"] and data["submitted_answers"]["y"] < 0:
+            data["format_errors"]["y"] = "Negative numbers are not allowed"
+
+    def grade(data):
+        if math.isclose(data["score"], 0.0) and data["submitted_answers"]["y"] > data["params"]["x"]:
+            data["partial_scores"]["y"]["score"] = 0.5
+            data["score"] = 0.5
+            data["feedback"]["y"] = "Your value for $y$ is larger than $x$, but incorrect."
+    ```
+
+=== "Example `info.json`"
+
+    ```json title="info.json"
+    {
+        "uuid": "cbf5cbf2-6458-4f13-a418-aa4d2b1093ff",
+        "title": "Doubling and Tripling",
+        "topic": "Basic Arithmetic",
+        "tags": ["sp24"],
+        "type": "v3",
+    }
+    ```
+
+### Elements
+
+Questions are written in HTML and **Elements**. Elements are pre-built interface elements for common questions, like numerical inputs, multiple choice, and more. **Custom Elements** are reusable components that can be tailored to your course and provide a way to create more complex questions. Custom elements are not necessary for most courses.
```

Comment:
More workshopping. As I mentioned elsewhere, I don't think that custom elements necessarily need to be mentioned here, especially given the "not necessary for most courses" callout.

```suggestion
Questions can use **elements** to accept student input, display diagrams, control how questions are displayed, and more. **Custom Elements** are reusable components that can be tailored to your course and provide a way to create more complex questions. Custom elements are not necessary for most courses.
```

Reply:
Looks good!